diff --git a/modules/agent/g/cfg.go b/modules/agent/g/cfg.go
index 1b0823d..e40f7ff 100644
--- a/modules/agent/g/cfg.go
+++ b/modules/agent/g/cfg.go
@@ -21,6 +21,8 @@ import (
 	"sync"

 	"github.com/toolkits/file"
+	"io/ioutil"
+	"strings"
 )

 type PluginConfig struct {
@@ -91,6 +93,29 @@ func Hostname() (string, error) {
 		return hostname, nil
 	}

+	if os.Getenv("ENDPOINT") != "" {
+		hostname = os.Getenv("ENDPOINT")
+		return hostname, nil
+	}
+
+	// parse /etc/endpoint.env ( ENDPOINT=xxx_xxx_1.2.3.4 )
+	filePath := "/etc/endpoint.env"
+	if _, err := os.Stat(filePath); err == nil {
+		data, _ := ioutil.ReadFile(filePath)
+		str := string(data)
+		if !strings.ContainsAny(str, "ENDPOINT") {
+			log.Panic("ERROR: /etc/endpoint.env missing ENDPOINT")
+		}
+		if !strings.ContainsAny(str, "=") {
+			log.Panic("ERROR: /etc/endpoint.env missing =")
+		}
+		str = strings.Trim((strings.SplitAfter(str, "ENDPOINT"))[1], " ")
+		hostname = strings.Trim((strings.SplitAfter(str, "="))[1], " ")
+		if len(hostname) > 1 {
+			return hostname, nil
+		}
+	}
+
 	hostname, err := os.Hostname()
 	if err != nil {
 		log.Println("ERROR: os.Hostname() fail", err)
diff --git a/modules/agent/main.go b/modules/agent/main.go
index 0b0c16b..82e9041 100644
--- a/modules/agent/main.go
+++ b/modules/agent/main.go
@@ -50,6 +50,7 @@ func main() {
 		g.InitLog("info")
 	}

+	os.Setenv("FALCON_AGENT_RUNTIME", g.VERSION)
 	g.InitRootDir()
 	g.InitLocalIp()
 	g.InitRpcClients()


diff --git a/modules/agent/funcs/dfstat.go b/modules/agent/funcs/dfstat.go
index 1a82699..296bea1 100644
--- a/modules/agent/funcs/dfstat.go
+++ b/modules/agent/funcs/dfstat.go
@@ -61,6 +61,10 @@ func DeviceMetrics() (L []*model.MetricValue) {
                        continue
                }

+               if du.FsVfstype == "iso9660" {
+                       continue
+               }
+
                diskTotal += du.BlocksAll
                diskUsed += du.BlocksUsed
